lib/ae_mdw/db/format.ex

```diff
@@ -286,31 +288,44 @@ defmodule AeMdw.Db.Format do
     |> update_in(["previous"], fn prevs -> Enum.map(prevs, &expand_name_info(state, &1)) end)
   end
 
+  defp custom_encode(state, :channel_close_mutual_tx, tx, _tx_rec, _signed_tx, _txi, _block_hash) do
+    put_channel_participants(tx, state)
+  end
+
   @payload_index 3
 
-  defp custom_encode(_state, :channel_close_solo_tx, tx, tx_rec, _signed_tx, _txi, _block_hash) do
-    tx_rec
-    |> elem(@payload_index)
-    |> put_channel_offchain_round(tx)
+  defp custom_encode(state, :channel_close_solo_tx, tx, tx_rec, _signed_tx, _txi, _block_hash) do
+    tx
+    |> put_channel_offchain_round(elem(tx_rec, @payload_index))
+    |> put_channel_participants(state)
+  end
+
+  defp custom_encode(
+         state,
+         :channel_force_progress_tx,
+         tx,
+         _tx_rec,
+         _signed_tx,
+         _txi,
+         _block_hash
+       ) do
+    put_channel_participants(tx, state)
   end
 
-  defp custom_encode(_state, :channel_slash_tx, tx, tx_rec, _signed_tx, _txi, _block_hash) do
-    tx_rec
-    |> elem(@payload_index)
-    |> put_channel_offchain_round(tx)
+  defp custom_encode(state, :channel_slash_tx, tx, tx_rec, _signed_tx, _txi, _block_hash) do
+    tx
+    |> put_channel_offchain_round(elem(tx_rec, @payload_index))
+    |> put_channel_participants(state)
```

Comment:
Can we use the already decoded value inside the tx for the channel pk? E.g.
```elixir
|> put_channel_participants(:channel_slash_tx.channel_pubkey(tx), state)
```

Reply:
With the `tx` map serialized by the node, we can get in a generic way for the 7 transaction types.

Reply:
On each `custom_encode` clause we know the transaction type and already do different things per transaction type. I think it will look better and it will type-check once we get the aeternity types in MDW. But you don't need to change it if you don't want to

Reply:
The type received is not the one with the getter, i.e `:channel_slash_tx.channel_pubkey(tx)` wouldn't work. There would be a need for additional conversion.

Reply:
Done using `signed_tx` with `aesc_utils`. As it was already assured to be safe by the Node, I would consider it a micro optimization but it will be fun to play around with this.